source/iface/math-functions.rst

```diff
@@ -7,3 +7,2731 @@
+``__swizzled_vec__`` type, in which case the return type is the
+corresponding ``sycl::vec``.
+
+===============
+``sycl::ilogb``
+===============
+
+.. rubric:: Overloads 1-3
+
+::
+
+  int ilogb(float x);
+  int ilogb(double x);
+  int ilogb(half x);
+
+Compute the integral part of :math:`log_r(|x|)` and return
+the result as an integer, where ``r`` is the value returned
+by ``std::numeric_limits<decltype(x)>::radix``.
+
+.. rubric:: Overload 4
+
+::
+
+  template<typename NonScalar>
+  /*return-type*/ ilogb(NonScalar x);
+
+Available only if all of the following conditions are met:
+
+* ``NonScalar`` is ``sycl::marray``, ``sycl::vec``, or the
+  ``__swizzled_vec__`` type;
+* The element type is ``float``, ``double``, or ``half``.
+
+For each element of ``x``, compute the integral part of
+:math:`log_r(|x[i]|)` and return the result as an integer,
+where ``r`` is the value returned by
+``std::numeric_limits<NonScalar::value_type)>::radix``.
+
+The return type depends on ``NonScalar``. If ``NonScalar``
+is ``sycl::marray``, the return type is ``sycl::marray`` of
+``int`` with the same number of element as ``NonScalar``. If
+``NonScalar`` is ``sycl::vec`` or the ``__swizzled_vec__`` type,
+the return type is ``sycl::vec`` of ``int32_t`` with the same
+number of elements as ``NonScalar``.
+
+===============
+``sycl::ldexp``
+===============
+
+.. rubric:: Overloads 1-3
+
+::
+
+  float ldexp(float x, int k);
+  double ldexp(double x, int k);
+  half ldexp(half x, int k);
+
+Returns the value ``x`` multiplied by :math:`2^k`.
+
+.. rubric:: Overload 4
+
+::
+
+  template<typename NonScalar1, typename NonScalar2>
+  /*return-type*/ ldexp(NonScalar1 x, NonScalar2 k);
+
+Available only if all of the following conditions are met:
+
+* ``NonScalar1`` is ``sycl::marray``, ``sycl::vec``, or
+  the ``__swizzled_vec__`` type;
+* The element type of ``NonScalar1`` is ``float``,
+  ``double``, or ``half``;
+* If ``NonScalar1`` is ``sycl::marray``, ``NonScalar2`` is
+  ``sycl::marray`` of int with the same number of elements
+  as ``NonScalar1``; and
+* If ``NonScalar1`` is ``sycl::vec`` or the ``__swizzled_vec__``
+  type, ``NonScalar2`` is ``sycl::vec`` or the ``__swizzled_vec__``
+  type of ``int32_t`` with the same number of elements as ``NonScalar1``.
+
+
+Returns the value ``x[i]`` multiplied by :math:`2^{k[i]}`
+for each element of ``x`` and ``k``.
+
+The return type is ``NonScalar1`` unless ``NonScalar1`` is the
+``__swizzled_vec__`` type, in which case the return type is the
+corresponding ``sycl::vec``.
+
+.. rubric:: Overload 5
+
+::
+
+  template<typename NonScalar>
+  /*return-type*/ ldexp(NonScalar x, int k);
+
+Available only if all of the following conditions are met:
+
+* ``NonScalar`` is ``sycl::marray``, ``sycl::vec``, or the
+  ``__swizzled_vec__`` type;
+* The element type is ``float``, ``double``, or ``half``.
+
+Returns the value ``x[i]`` multiplied by :math:`2^k`
+for each element of ``x`` and ``k``.
```

Comment:
I don't think "for each element of ... k" makes sense in this overload.  It should be more like "2^k multiplied by each element of ..."